box: add is_sync option (#8907)
tarantool/tarantool: changelogs/unreleased/gh-8650-add-is_sync-option.md

## feature/box

> * Added a new parameter `is_sync` to `box.begin()` and `box.commit()`.
> Thanks to this parameter can make the transaction synchronous. To do this
> set `is_sync = true`. `box.commit()` cannot change `is_sync` to `false`
> if the transaction was opened as synchronous or has writes to synchro
> spaces.
> 
> * Also, now for these purposes can use `box.atomic()` with the `is_sync`
> parameter (gh-8650).

Reviewer: ```suggestion
* Added the new `is_sync` parameter to `box.begin()`, `box.commit()`, and `box.atomic()`.
To make the transaction synchronous, set the `is_sync` option to `true`. `box.commit()` can't change `is_sync` to `false`
if the transaction was opened as synchronous or has writes to synchronous
spaces (gh-8650).
```

Author: Fixed.

> ## feature/box

Reviewer 2: The docbot only mentions `box.atomic`, but you added it to `box.begin` and `box.commit` too. Please, document all three. Same for the netbox changes.

Author: ### **What have we agreed:**

-  We leave `is_sync` in all three places (begin, commit, atomic) with the only `true` option.
    - We throw an error to the user when he specifies `is_sync != true` or any other value like `is_sync = "some string"`.

-  For users, we document only `atomic`. Documentation for the guru about `begin` and `commit` will be described only in the commit message.

- `is_sync`  should also appear in the `iproto` protocol (`iproto_begin` and `iproto_commit`).

> * Added the new `is_sync` parameter to `box.atomic()`. To make the

Author: Fixed.

transaction synchronous, set the `is_sync` option to `true`. Setting

> `is_sync = false` is prohibited. If to set any value other than true/nil,

Author: Fixed.

> ## feature/box

Author: What have we agreed:

-  We leave `is_sync` in all three places (begin, commit, atomic) with the only `true` option.
    - We throw an error in the face of the user when he specifies `is_sync = false` or any other value like `is_sync = "some string"`.

-  For users, we document only `atomic`. Documentation for the guru about `begin` and `commit` will be described only in the commit message.

- `is_sync`  should also appear in the `iproto` protocol (`iproto_begin` and `iproto_commit`).
